Add 'Items Merge' task for 'IO' module

danikk812 opened the pull request.

danikk812 committed:
Add 'Items Merge' task for 'IO' module

dark11ght commented on src/com/dmdev/javacore/io/csv/items/util/ItemsCSVUtil.java:
```diff
@@ -0,0 +1,77 @@
+
+    private ItemsCSVUtil() {
+    }
+
```
к коменту выше про ошибки 

```
 public static void createCSV(Path path, String content) {
        try {
            Files.write(path, Collections.singleton(content.trim()));
        } catch (IOException e) {
            throw new RuntimeException("createCSV() oops something wrong " + e, e);
        }
    }
```

dark11ght commented on src/com/dmdev/javacore/io/csv/items/CSVMergeRunner.java:
```diff
@@ -0,0 +1,58 @@
+            ItemsCSVUtil.saveErrorsCSV(itemsErrorData, errorsFilePath);
+
+            System.out.println("Items merge completed successfully.");
+
```
лучше обратотаь ошибки непосредственно в месте возникновения. например

```
Exception in thread "main" java.lang.RuntimeException: createCSV() oops something wrong java.nio.file.NoSuchFileException: src/com/dmdev/javacore/io/csv/items/resources/items-name.csv
	at com.dmdev.javacore.io.csv.items.util.ItemsCSVUtil.createCSV(ItemsCSVUtil.java:19)
	at com.dmdev.javacore.io.csv.items.CSVMergeRunner.main(CSVMergeRunner.java:39)
Caused by: java.nio.file.NoSuchFileException: src/com/dmdev/javacore/io/csv/items/resources/items-name.csv
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:484)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:228)
	at java.base/java.nio.file.Files.write(Files.java:3581)
	at java.base/java.nio.file.Files.write(Files.java:3633)
	at com.dmdev.javacore.io.csv.items.util.ItemsCSVUtil.createCSV(ItemsCSVUtil.java:17)
	... 1 more
```

более читабельнее и понятнее чем

`An error occurred while files processing src/com/dmdev/javacore/io/csv/items/resources/items-name.csv`

dark11ght commented on src/com/dmdev/javacore/io/csv/items/util/ItemsCSVUtil.java:
```diff
@@ -0,0 +1,77 @@
+                    .skip(1)
+                    .map(line -> line.split(","))
+                    .collect(Collectors.toMap(
+                            arr -> arr[0],
```
Я бы вынес индекс в константу. так непонятно что за 0 или 1. типа так:

```
  private static final int ID = 0;
  private static final int PRICE = 1;
```

dark11ght commented on src/com/dmdev/javacore/io/csv/items/util/ItemsCSVUtil.java:
```diff
@@ -0,0 +1,77 @@
+
+    public static void saveResultCSV(Map<String, String> mergedItemsData, Path path) throws IOException {
+        List<String> lines = new ArrayList<>();
+        lines.add("ID,NAME,PRICE");
```
это можно вынести в константу доустим HEADER
`private static final String HEADER = "ID,NAME,PRICE";`

danikk812 committed:
Add src/com/dmdev/javacore/io/csv/items/resources/ items related files

danikk812 committed:
Apply review fixes to javacore.io.csv.items

Extract CSV files column indexes to constants
Improve error handling by catching errors directly at the point of origin
